### PR TITLE
SSCS-10889 Improve logging

### PR DESCRIPTION
**Pull request by filippo-possenti-version1**
### JIRA link (if applicable) ###
[SSCS-10889](https://tools.hmcts.net/jira/projects/SSCS/issues/SSCS-10889)

### Change description ###
- New log message when an incorrect benefit/issue code combination is used, intended to clarify what would otherwise cause a NullPointerException soon afterwards.
- Update of snakeyaml library version to fix a CVE
- Removal of unnecessary CVE suppressions
- Assessment on remaining CVE with related update of suppression date and reason

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X ] No
```

### CODE REVIEW

**Review by dattipoeHMCTS — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -427,10 +427,9 @@ dependencyManagement {
       entry 'spring-messaging'
     }
 
-  //CVE-2022-25857
-  dependencySet(group: 'org.yaml', version: '1.31') {
+    dependencySet(group: 'org.yaml', version: '1.31') {
```
Comment (hgulv1) [inline build.gradle:430]: there is no increment of snakeyaml it just looks like spacing/formatting changed here?

**Review by hgulv1 — CHANGES_REQUESTED** [review]
there is no increment of snakeyaml it just looks like spacing/formatting changed here?

File: build.gradle (hunk 1 above)
Comment (filippo-possenti-version1) [reply build.gradle:430]: Yes. Just aligning it with the entries above.

**Review by hgulv1 — APPROVED** [review]
all good

**Pull request merged**